$NetBSD: patch-aa,v 1.3 2015/12/29 23:34:56 dholland Exp $

Allow setting the prelude path from outside.

This is not the only scheme interpreter, so change the init filename
and similar strings accordingly.

--- steering.c.orig	Sun Jan 21 22:15:53 1996
+++ steering.c
@@ -62,7 +62,11 @@
 
 #define TOP_LEVEL_PROMPT "\n==> "
 
+#include "path.h"
+
+#ifndef STANDARD_PRELUDE_PATHNAME
 #define STANDARD_PRELUDE_PATHNAME "/usr/local/lib/scheme/prelude.scheme"
+#endif
 
 Private String OPENING  =
 "Welcome to UMB Scheme, version      Copyright (c) 1988,1996 William R Campbell.\n\
@@ -91,10 +95,10 @@
 Private void Steering()
 {
 	Character Opening[400];
-	String	Init_Filename = getenv ("SCHEME_INIT");
+	String	Init_Filename = getenv ("UMB_SCHEME_INIT");
 	Character Dot_Scheme_Filename [256];
 	sprintf (Opening, "%s", OPENING);
-	sprintf (Dot_Scheme_Filename, "%s/.scheme", getenv ("HOME"));
+	sprintf (Dot_Scheme_Filename, "%s/.umb-scheme", getenv ("HOME"));
 
 	Initializations();
 
